torch/distributed/elastic/multiprocessing/api.py

```diff
@@ -670,9 +670,13 @@ def _poll(self) -> Optional[RunProcsResult]:
             if self._is_done():
                 # we should ALWAYS have ALL the return values when all the processes are done
                 self._worker_finished_event.set()
-                # Wait untill all processes are finished. At this point workers finished executing
-                # user function
-                self._pc.join()
+
+                # At this point workers finished running the user function
+                # But the child process might still have not exited. Wait for them.
+                # pc.join() blocks [forever] until "a" proc exits. Loop until all of them exits.
+                while not self._pc.join():
```

Comment:
should we have a timeout on this? wondering what happens if we have a dead/hung worker process?

Reply:
When you reach this line, we've already validated that either: 

1. the entrypoint function actually ran and returned a result
2. -- or -- at least one of the child procs have failed (and a SIGTERM was sent to the rest)

We're waiting for the spawned child proc to exit after the user-provided function has already returned.
This potentially could hang but we were waiting for `_pc.join()` indefinitely before this change as well.

Reply:
got it -- sgtm